### ISSUE TITLE
Remove built-in zoom control 

### ISSUE THREAD

**Issue by kidwm** [react-d3/react-d3-map#18]
<Map /> should expose scale state and functions or always get scale from props.
and provide an overlay layer to place zoom control or other elements.

**Comment by chilijung** [comment]
https://github.com/react-d3/react-d3-map#with-zoomcontrol

We remove default zoomControl in v0.8.0 so you could build your own zoomControl :)

**Comment by chilijung** [comment]
@kidwm and also popup event trigger has support in v0.8.0

example:

``` js
var onPolygonMouseOut= function(e, dom , d, i) {
    console.log('out')
    e.hidePopup();
  }
  var onPolygonMouseOver= function(e, dom, d, i) {
    console.log('over')
    e.showPopup();
  }
  var onPolygonClick= function(e, dom, d, i) {
    // e.showPopup();
  }
  var onPolygonCloseClick= function(e, id) {
    // e.hidePopup();
  }
  var popupContent = function(d) { return 'hi, i am polygon'; }

  var Container = React.createClass({
    render: function() {
      return (
        <g>
          <PolygonGroup
            key= {"polygon-test"}
            data= {data}
            popupContent= {popupContent}
            onClick= {onPolygonClick}
            onCloseClick= {onPolygonCloseClick}
            onMouseOver= {onPolygonMouseOver}
            onMouseOut= {onPolygonMouseOut}
            polygonClass= {"your-polygon-css-class"}
          />
        </g>
      )
    }
  })
```

**Comment by kidwm** [comment]
@chilijung does the `e` mean element?
if so, maybe you can consider removing the `dom` parameter, and get it from element.findDOMNode or something alike else.

**Comment by chilijung** [comment]
@kidwm e means event, dom means which element you are triggering. Or maybe I should integrate `dom` into `e.dom`, how do you think?

**Comment by kidwm** [comment]
@chilijung I just think that it is strange to bind showPopup() and hidePopup() on an event.
There should be a Popup component to do such things.

BTW, event.dom looks better for me.

**Comment by chilijung** [comment]
@kidwm Oops, e means the element you are trigger yeah... Sorry. Ok I'll put dom into `element.dom`, sounds better? Thanks

**Comment by chilijung** [comment]
@kidwm in v0.8.1 version you can get component dom via 

``` js
var onMarkerMouseOut= function(component, d, i) {
    component.hidePopup();
    // to get the dom
    var dom = ReactDOM.findDOMNode(component.getDOM);
  }
```

**Comment by kidwm** [comment]
@chilijung getDOM should be a function but not a property, since get is a verb.
Maybe you could name it domRef or something else.

**Comment by chilijung** [comment]
fix in v0.8.2, use domRef instead.
